src/analyze/analyze.c

```diff
@@ -2145,7 +2226,19 @@ static int do_condition(int argc, char *argv[], void *userdata) {
 }
 
 static int do_verify(int argc, char *argv[], void *userdata) {
-        return verify_units(strv_skip(argv, 1), arg_scope, arg_man, arg_generators);
+        int r;
+        _cleanup_strv_free_ char **filenames = NULL;
+        _cleanup_(rm_rf_and_freep) char *tempdir = NULL;
+
+        r = setup_working_dir(&tempdir);
+        if (r < 0)
+                return log_error_errno(r, "Failed to setup working directory");
+
+        r = process_aliases(argv, tempdir, &filenames);
+        if (r < 0)
+                return log_error_errno(r, "Couldn't process aliases");
```

Comment:
you already log about all errors in process_aliases(). If you log again here, then you will log twice, which is confusing.

In systemd functions should either log about all errors on their own (if they are "main-function-like") or about none (if they are "library-function-like"). In the latter case they should return an error code so that the caller can log (or propagate the error further). The line between the two is a bit blurry, but the gist is: figure out what kind of function your function is, and then either log about all errors or about none, and then make sure that something further up does the logging or doesn't, so that every error is logged about once.

(log_debug() level logging is always ok though and doesn't really count as logging in this context)